### DISCUSSION TITLE
"No Spaceport" in planet list UI

### DISCUSSION THREAD

**Discussion by brendanjones** [endless-sky/endless-sky#7169] [Balance, Mechanics, and Interactions]
For most planets, the top line in the map UI says if the planet is friendly/hostile/restricted/dominated. But if the planet is not populated, it says "No Spaceport". This doesn't make sense to me, for a lack of a spaceport doesn't necessarily mean a planet is unpopulated, and it has nothing to do with how much a planet likes you.

To maintain consistency with that line being about your reputation with the planet, wouldn't it be better if instead of "No Spaceport" it said "Unpopulated", or "No Government"?

If we think it important to say if there's a spaceport or not, that could go on the next line as a boolean value, like Outfitter and Shipyard. 

<img width="273" alt="Screen Shot 2022-08-23 at 19 57 44" src="https://user-images.githubusercontent.com/16049594/186586874-59c863fd-0083-43b6-84d0-d6608017b69f.png">

**Comment by mOctave** [comment]
If there is no spaceport, then there are no spaceport authorities who may or may not let you land.

You could land on a planet with no spaceport and still find yourself on a civilized world. It doesn't matter what the attitude of the government that owns the planet is as long as you can land, unless someone writes a mission saying otherwise. What that line really says is one of five things:

- __Friendly:__ The spaceport authorities are happy to let you land.
- __Restricted:__ You are in good standing with the planetary government, but do not have the clearance to land on the planet.
- __Hostile:__ The spaceport authorities don't like you, and refuse to let you land.
- __Dominated:__ The spaceport authorities do not dare refuse to allow you to land, regardless of your reputation with the government.
- __No Spaceport:__ You can land if you want – there's no one stopping you. Just don't expect to be able to refuel, get jobs, or improve your fleet while you're on the surface.

**Reply by MasterOfGrey** [reply]
Basically this yeah - I'm not sure if there's a better way to communicate that information though?

**Reply by brendanjones** [reply]
Yes I get all that. My point is that the first four tell you your level of access to the planet/their position towards you, while “No spaceport” has nothing to do with level of access so should either be changed, or not be included on this line in the map UI.

**Reply by mOctave** [reply]
There might be a better way to phrase it than "no spaceport," but all the ones I can think of are either potentially inaccurate or are long-winded and confusing.

**Reply by MasterOfGrey** [reply]
> while “No spaceport” has nothing to do with level of access 

I mean, it explicitly does, you can access it freely cause there's nothing to stop you or aid you either way.
I do agree it would be nice if we could get some words that communicate the information in a way that has less possible cause for confusion though.

**Reply by brendanjones** [reply]
> I mean, it explicitly does

Sorry but I think you're thinking based on your knowledge of game mechanics, not the actual concept of what a spaceport is. Put yourself in the shoes of a new player who also hasn't played any EV: what does the existence of a building for welcoming spaceships have to do with what the people of the planet think of you? That's what the other four statuses are, they're what the people of the planet think of you. What they think determines if they allow you to land or not.

If anything 'No spaceport' could mean that you can't land on the planet because there's no nice landing pad waiting for you at the spaceport. Except you can land on these planets.

**Reply by mOctave** [reply]
> Put yourself in the shoes of a new player who also hasn't played any EV: what does the existence of a building for welcoming spaceships have to do with what the people of the planet think of you?

I think maybe we're disagreeing on what a spaceport is.  
Analogy: you can land a plane in a variety of places. You might get into trouble later, but you can still land. If you land at an airport, however, you have to follow the directions of air traffic control and the airport may refuse you access to land. If you land anyways, airport security is probably going to arrest you. Pilots don't land without permission at an airport because if they do they may never fly again. On the other hand, people smuggle things all the time in places without official airports.

**Reply by brendanjones** [reply]
No that's exactly how a spaceport functions, no disagreement there. And yet, whether a spaceport exists or not has no bearing on whether the inhabitants of the planet want you to land, and as you say in your example, a lack of spaceport doesn't stop a spaceship landing. Therefore, to repeat, 'No spaceport' doesn't belong in the same list as friendly/hostile/restricted/dominated.

It's the difference between "where on the planet do I land" and "do the people of this planet want/allow me to land".
